docs/HARDWARE-VERIFICATION.md

```diff
@@ -0,0 +1,72 @@
+2. Burn the resulting ```verification.raw``` image file onto a USB stick and inserting it into the edge node.
+3. Have the machine boot from USB from BIOS.
+
+To produce the verification.raw image, we can execute the command ```docker run lfedge/eve-verification verification_raw > verification.raw```
+
+In cases such as using HPE iLO or Dell DRAC your BIOS will not be able
+to boot from a disk-based image, so we need an ISO image. The only
+difference is in step #1 that then becomes ```docker run lfedge/eve-verification verification_iso > verification.iso```.
+
+## Running the verification image via iPXE
+
+[iPXE](https://en.wikipedia.org/wiki/IPXE) is a modern Preboot eXecution Environment and
+a boot loader that allows operating system images to be downloaded right at the moment
+of booting (checkout [deployment](DEPLOYMENT.md)).
+To get the necessary images required by iPXE, we can execute the command
+```docker run lfedge/eve-verification verification_net | tar xf -```. Apart from the image, iPXE
+expects a configuration file at certain URLs in order to proceed with the boot process.
+Here's an example of a iPXE configuration file used to run an EVE-OS
+verification image locally in a hardware lab. Note the changes made to both ```url``` and ```eve_args``` variables.
+
+```console
+#!ipxe
+# set url https://github.com/lf-edge/eve/releases/download/snapshot/amd64.
+set url https://10.0.0.2/eve/releases/download/snapshot/amd64.
+# set eve_args eve_soft_serial=${ip} eve_reboot_after_install
+set eve_args eve_soft_serial=${ip} eve_install_server=zedcontrol.hummingbird.zededa.net eve_reboot_after_install
+
+# you are not expected to go below this line
+set console console=ttyS0 console=ttyS1 console=ttyS2 console=ttyAMA0 console=ttyAMA1 console=tty0
+set installer_args root=/initrd.image find_boot=netboot overlaytmpfs fastboot
+
+# you need to be this ^ tall to go beyond this point
+kernel ${url}kernel ${eve_args} ${installer_args} ${console} ${platform_tweaks} initrd=amd64.initrd.img initrd=amd64.verification.img initrd=amd64.initrd.bits initrd=amd64.rootfs.img
+initrd ${url}initrd.img
+initrd ${url}verification.img
+initrd ${url}initrd.bits
+initrd ${url}rootfs.img
+boot
+```
+
+## Logs of the verification process
+
+The verification images uses multiple utilities to gather the logs during
+the verification process. Among them we find: cpuinfo, meminfo, dmesg, smartctl,
+lsblk, lspci, lsusb, scsi, the hardware model as described in
+[HARDWARE-MODEL](./HARDWARE-MODEL.md), etc. Additionally, it stores the
+results of testing a static and a dynamic (i.e., using DHCP) network
+configuration and the read performance of the available storage devices
+using [fio](https://github.com/axboe/fio).
+
+## Getting the results of the verification process
+
+The verification image stores the results of all tests in the *invertory
```

Comment:
The inventory partition is persistent, so we can boot the device with the installed EVE after an iPXE boot and extract the tests and the verification data. However, I plan to improve this part so that during the iPXE boot and after the verification process is done, we send all the data also via HTTP.